content/learning-paths/mobile-graphics-and-gaming/android_halide/processing-workflow.md

```diff
@@ -0,0 +1,585 @@
+---
+# User change
+title: "Building a Simple Camera Image Processing Workflow"
+
+weight: 3
+
+layout: "learningpathall"
+---
+
+## Objective
+In this section, we will build a real-time camera processing pipeline using Halide. First, we capture video frames from a webcam using OpenCV, then implement a Gaussian (binomial) blur to smooth the captured images, followed by thresholding to create a clear binary output highlighting prominent image features. After establishing this pipeline, we will measure performance and then explore Halide’s scheduling options—parallelization and tiling—to understand when they help and when they don’t.
+
+## Gaussian blur and thresholding
+Create a new camera-capture.cpp file and modify it as follows:
+```cpp
+#include "Halide.h"
+#include "HalideRuntime.h"   // for Runtime::Buffer make_interleaved
+#include <opencv2/opencv.hpp>
+#include <iostream>
+#include <string>
+#include <cstdint>
+#include <exception>
+
+using namespace cv;
+using namespace std;
+
+// Clamp coordinate within [0, maxCoord - 1].
+static inline Halide::Expr clampCoord(Halide::Expr coord, int maxCoord) {
+    return Halide::clamp(coord, 0, maxCoord - 1);
+}
+
+int main() {
+    // Open the default camera.
+    VideoCapture cap(0);
+    if (!cap.isOpened()) {
+        cerr << "Error: Unable to open camera." << endl;
+        return -1;
+    }
+
+    while (true) {
+        // Capture frame (typically interleaved BGR).
+        Mat frame;
+        cap >> frame;
+        if (frame.empty()) {
+            cerr << "Error: Received empty frame." << endl;
+            break;
+        }
+        if (!frame.isContinuous()) frame = frame.clone();
+
+        const int width    = frame.cols;
+        const int height   = frame.rows;
+        const int channels = frame.channels(); // 3 (BGR) or 4 (BGRA)
+
+        // Wrap the interleaved OpenCV frame for Halide.
+        auto in_rt = Halide::Runtime::Buffer<uint8_t>::make_interleaved(
+            frame.data, width, height, channels);
+        Halide::Buffer<> inputBuffer(*in_rt.raw_buffer()); // front-end view
+
+        // Define ImageParam (x, y, c) and declare interleaved layout.
+        Halide::ImageParam input(Halide::UInt(8), 3, "input");
+        input.set(inputBuffer);
+        input.dim(0).set_stride(channels); // x-stride = C (interleaved)
+        input.dim(2).set_stride(1);        // c-stride = 1 (adjacent bytes)
+        input.dim(2).set_bounds(0, channels);
+
+        // Spatial vars.
+        Halide::Var x("x"), y("y");
+
+        // Grayscale in Halide
```

Comment:
The implementation of color conversion should be consistent with other section, i.e. fusion.md